src/expr/src/relation/func.rs

```diff
@@ -543,33 +543,32 @@ fn jsonb_array_elements<'a>(a: Datum<'a>, temp_storage: &'a RowArena, stringify:
 }
 
 fn regexp_extract(a: Datum, r: &AnalyzedRegex) -> Option<Row> {
-    let mut row_packer = repr::RowPacker::new();
-    match a {
-        Datum::String(s) => {
-            let r = r.inner();
-            let captures = r.captures(s)?;
-            let datums_iter = captures
-                .iter()
-                .skip(1)
-                .map(|m| Datum::from(m.map(|m| m.as_str())));
-            Some(row_packer.pack(datums_iter))
-        }
-        _ => None,
-    }
+    let r = r.inner();
+    let captures = r.captures(a.unwrap_str())?;
```

Comment:
Stylistically I’d slightly prefer the unwrap to be on its own line (`let s = a.unwrap_str()`), on the principle that “panics should jump out at you while skimming”

Reply:
👍 done